src/QlikConnectorPSExecute/UserInteractive.cs

```diff
@@ -0,0 +1,346 @@
+                            res = LsaAddAccountRights(policyHandle, sid, userRights, 1);
+
+                        winErrorCode = LsaNtStatusToWinError(res);
+                        if (winErrorCode != 0)
+                        {
+                            errorMessage = $"LsaAddAccountRights failed: {winErrorCode}";
+                            throw new Win32Exception((int)winErrorCode, errorMessage);
+                        }
+                    }
+                    finally
+                    {
+                        LsaClose(policyHandle);
+                    }
+                }
+                FreeSid(sid);
+            }
+        }
+    }
+    #endregion
+
+    // Local security rights managed by the Local Security Authority
+    public class LocalSecurityAuthorityRights
+    {
+        // Log on as a service right
+        public const string LogonAsService = "SeServiceLogonRight";
+        // Log on as a batch job right
+        public const string LogonAsBatchJob = "SeBatchLogonRight";
+        // Interactive log on right
+        public const string InteractiveLogon = "SeInteractiveLogonRight";
+        // Network log on right
+        public const string NetworkLogon = "SeNetworkLogonRight";
+        // Generate security audit logs right
+        public const string GenerateSecurityAudits = "SeAuditPrivilege";
+    }
+
+    /* added wrapper for PowerShell */
+    public class InteractiveUser : IDisposable
+    {
+        #region Variables & Properties
+        private static bool IsLocallyDomainUser { get; set; }
+        private static bool IsLocalUser { get; set; }
+        private static NTAccount AccountInfo { get; set; }
+        private static string CurrentRight { get; set; }
+        #endregion
+
+        #region Constructor
+        public InteractiveUser(NTAccount accountInfo)
+        {
+            if(String.IsNullOrEmpty(CurrentRight))
+              CurrentRight = LocalSecurityAuthorityRights.InteractiveLogon;
+
+            AccountInfo = accountInfo;
+
+            IsLocalUser = IsLocalWinUser();
+            if (!IsLocalUser)
+            {
+                var results = GetRights();
+                if (!results.Contains(CurrentRight))
+                {
+                    AddRight(CurrentRight);
+                    IsLocallyDomainUser = false;
+                }
+                else
+                    IsLocallyDomainUser = true;
+            }
+        }
+
+        public void Dispose()
+        {
+            if(!IsLocallyDomainUser && !IsLocalUser)
+            {
+                RemoveRight(CurrentRight);
+            }
+        }
+        #endregion
+
+        #region Static Methods
+        private static bool IsLocalWinUser()
+        {
+            string strMachineName = System.Environment.MachineName;
+            return WindowsIdentity.GetCurrent().Name.ToUpper().Contains(strMachineName.ToUpper());
+        }
+
+        private static IList<string> GetRights()
```

Comment:
@mossbeard 
why not a private static LocalSecurityControler, why generating in each functiona new controller